src/geometry/MeshFragSceneNode.cxx

```diff
@@ -169,61 +97,42 @@ void MeshFragSceneNode::Geometry::render()
     // set the color
     sceneNode->setColor();
 
-    if (list != INVALID_GL_LIST_ID)
-        glCallList(list);
+    if (BZDBCache::lighting)
+    {
+        if (BZDBCache::texture)
+            vboVTN.enableArrays();
+        else
+            vboVTN.enableArrays(false, true, false);
+    }
     else
     {
-        if (BZDBCache::lighting)
-        {
-            if (BZDBCache::texture)
-                drawVTN();
-            else
-                drawVN();
-        }
+        if (BZDBCache::texture)
+            vboVTN.enableArrays(true, false, false);
         else
-        {
-            if (BZDBCache::texture)
-                drawVT();
-            else
-                drawV();
-        }
+            vboVTN.enableArrays(false, false, false);
```

Comment:
I think this block would be clearer if you set variables for the call to `enableArrays` based on the various state values and only made one call. As it is, the various boolean literal values have no meaning to anyone without finding the function declaration and figuring out what they mean.

Reply:
Named parameter. We have in Ada, why they are not in c++

Reply:
good question. Python has them too. I guess no language can be perfect 😁

Reply:
for now I have defined a enableVertexOnly for handling this case (for shadow mainly)
for the normal case enableArrays without parameters enable all buffers that are defined for the object.
I hope the code is more clear now